File: LoginPage2/app/src/main/java/com/example/loginpage/MainActivity.java
package com.example.loginpage;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.squareup.picasso.Picasso;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String url = getString(R.string.url);
        ImageView imagen = findViewById(R.id.imagen);
        Picasso.get().load(url).into(imagen);
        final Context context = getApplicationContext();
        final Button boton = findViewById(R.id.botonlogin);
        boton.setOnClickListener(
                new View.OnClickListener() {
                    public void onClick(View v) {
                        EditText user = findViewById(R.id.usuario);
                        String usertexto = user.getText().toString();
                        EditText pass = findViewById(R.id.password);
                        String passtexto = pass.getText().toString();

                        if(!passtexto.isEmpty() && !usertexto.isEmpty()) {
                            Intent intent = new Intent(MainActivity.this, Home.class);
                            startActivity(intent);
                        } else {
                            Toast toast = Toast.makeText(context, getString(R.string.fallo), Toast.LENGTH_SHORT);
                            toast.show();
                        }
                    }
                }
        );

        final Button botonregistro = findViewById(R.id.botonregistro);
        botonregistro.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, RegistroUsuario.class);
                startActivity(intent);
            }
        });
    }
}

File: Kotlin5/app/src/main/java/cl/desafiolatam/desafiounobase/MainActivity.kt
package cl.desafiolatam.desafiounobase

import android.content.Intent
import android.content.SharedPreferences
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.constraintlayout.widget.ConstraintLayout
import com.google.android.material.snackbar.Snackbar
import com.google.android.material.textfield.TextInputEditText

class MainActivity : AppCompatActivity() {
    companion object {
    	private const val base = "SP"
	const val user = base + "user"
    }
    lateinit var nameInput: TextInputEditText
    lateinit var advance: Button
    lateinit var container: ConstraintLayout
    lateinit var sharedPreferences: SharedPreferences
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
	val file = "cl.desafiolatam.sharedpreferences"
	sharedPreferences = getSharedPreferences(file, Context.MODE_PRIVATE)
        nameInput = findViewById(R.id.name_input)
        advance = findViewById(R.id.login_button)
        container = findViewById(R.id.container)
        setUpListeners()
    }

    private fun setUpListeners() {
        advance.setOnClickListener {
            if (nameInput.text!!.isNotEmpty()) {
                val intent: Intent
                if (hasSeenWelcome()) {
                    intent = Intent(this, HomeActivity::class.java)
                } else {
                    intent = Intent(this, WelcomeActivity::class.java)
                }
                startActivity(intent)
            } else {
                Snackbar.make(container, "El nombre no puede estar vacío", Snackbar.LENGTH_SHORT).show()
            }
        }
    }

    private fun hasSeenWelcome(): Boolean {
	val saved = sharedPreferences.getString(user, "").toString()
	if (saved == nameInput.text.toString()) {
		return true
	} else {
		sharedPreferences.edit().putString(user, nameInput.text.toString()).apply()
        	return false
        }
    }
}
